### PR TITLE
Added functionality to swap origin and destination button

### PR DESCRIPTION
**Pull request by AconexOfficial**
### Description

Added functionality to swap origin and destination button

## Checklist

- [x] My code follows the style guidelines
- [ ] I have documented my code
- [x] My changes generate no new warnings
- [x] I have linked the Issue
- [x] The Pull Request covers all the Checkpoints from the tagged Issue
- [x] I have noted the hours of work i have put in the issue

## I have edited following Parts of the project

- [x] Frontend
- [ ] Backend
- [ ] Github
- [ ] Root

## If tests are required did you make them?

- [x] No tests required
- [ ] Yes
- [ ] No

### CODE REVIEW

File: frontend/src/components/LocationSearch/LocationSearchInput.tsx [hunk 1]
```diff
@@ -22,9 +22,11 @@ import { LocationSearchList } from "./LocationSearchList";
 
 export interface LocationSearchInputProps {
   onSelectedLocationChanged: (location: Location) => void;
+  onInputGiven: (input: string) => void;
```
Comment (Paulomart) [inline frontend/src/components/LocationSearch/LocationSearchInput.tsx:25]:
```suggestion
  onSearchInputChanged: (input: string) => void;
```

File: frontend/src/components/LocationSearch/LocationSearchInput.tsx [hunk 2]
```diff
@@ -22,9 +22,11 @@ import { LocationSearchList } from "./LocationSearchList";
...
   selectedLocation: Location | null;
   inputLabel: string;
   prefixDataTestId?: string;
+  searchName?: string;
```
Comment (Paulomart) [inline frontend/src/components/LocationSearch/LocationSearchInput.tsx:29]: searchInput

File: frontend/src/components/LocationSearch/LocationSearchInput.tsx [hunk 3]
```diff
@@ -54,6 +57,11 @@ export const LocationSearchInput = (props: LocationSearchInputProps): JSX.Elemen
   const showLoadingIndicator = searchInput !== "" && (foundLocations.type === "outdated" || inputStillInDebounce);
   const showResults = searchInput !== "" && (foundLocations.type === "success" || foundLocations.type === "outdated");
 
+  const applyLocationName = (): void => {
+    setSearchInput(props.searchName ?? "");
```
Comment (Paulomart) [inline frontend/src/components/LocationSearch/LocationSearchInput.tsx:61]: remove line 34 as the state is now stored higher up.

**Review by Paulomart — CHANGES_REQUESTED** [review]

File: frontend/src/components/LocationSearch/LocationSearchInput.tsx (hunk 3 above)
Comment (AconexOfficial) [reply frontend/src/components/LocationSearch/LocationSearchInput.tsx]: Omitted due to time constraints

**Review by Paulomart — APPROVED** [review]

**Pull request merged**